programs/uxd/src/instructions/redeem.rs

```diff
@@ -160,24 +167,69 @@ struct DepositoryInfoForRedeem<'info> {
 
 pub(crate) fn handler(ctx: Context<Redeem>, redeemable_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
 
+    // Compute real outflow limit for this epoch (max of bps/amount options)
+    // Note: intermediary maths forced to use u128 to be able to multiply u64s safely
+    let outflow_limit_per_epoch_amount = std::cmp::max(
+        controller.outflow_limit_per_epoch_amount,
+        checked_as_u64(checked_div(
+            checked_mul(
+                checked_as_u128(controller.redeemable_circulating_supply)?,
```

Comment:
true! I still want to keep the enforcement here because: if we change the type to u64 one day, this might accidentally make this multiplication become u64*u64 which would be a problem, I prefer forcing everything onto u128 explicitely, and if one conversion is a no-op its no big deal IMO :)

I just like the explicitness about it: if we RELY on the behavior of u128, i prefer putting that in the code rather than letting the type inference figure it out

Reply:
ah sad, this throws a lint warning, I guess i'll just remove this for now

Reply:
If you want to enforce it for calculations, make `u128` explicit, such as:

```
checked_mul::<u128>(
      controller.redeemable_circulating_supply,
      checked_as_u128(controller.outflow_limit_per_epoch_bps)?,
 )?,
```